### PR TITLE
Feature pipeline parameters

### PR DESCRIPTION
**Pull request by fruttasecca**
This PR adds the concept of pipeline parameters. Pipeline parameters can be set up through the pipeline settings view when in the pipeline editor; moreover, pipeline parameters can be overridden for jobs.

- [x] The documentation reflects the changes.
- [x] I have manually tested the application to make sure the changes don’t cause any downstream issues.
- [x] In case I changed one of the services’ `models.py` I have performed the appropriate database migrations.

### CODE REVIEW

**Comment by yannickperrenet** [comment]
Resolves #125

File: orchest-sdk/python/orchest/parameters.py [hunk 1]
```diff
@@ -31,34 +32,44 @@ def get_params() -> Dict[str, Any]:
...
 
-    return params
+    return params, pipeline.get_params()
 
 
-def update_params(params: Dict[str, Any]) -> None:
-    """Updates the parameters of the current step.
+def update_params(
+    step_params: Optional[dict] = None, pipeline_params: Optional[dict] = None
+) -> None:
```
Comment (yannickperrenet) [inline orchest-sdk/python/orchest/parameters.py:40]: Didn't we want to add a warning as updating `pipeline_params` could introduce a race condition unaware to the user?

**Review by yannickperrenet — CHANGES_REQUESTED** [review]
I think we should add a warning somehow. Probably using the docstring:
```python
"""...
Warning:
    The warning message
"""
```
This will then also get rendered in the documentation as a warning using sphinx.

File: orchest-sdk/python/orchest/parameters.py (hunk 1 above)
Comment (fruttasecca) [reply orchest-sdk/python/orchest/parameters.py:40]: Will  do

**Comment by yannickperrenet** [comment]
When configuring the parameters of the job, you see something like:
```
Pipeline: pipe
    a:[5]
step-name
    b:[6]
```
but I am not sure whether this is clear enough to the user. The confusion to me is mostly introduced by the fact that in the top it already states:
```
Pipeline
pipe
```

Happy to hear your thoughts as maybe I am just imaging the confusion :)

**Review by yannickperrenet — APPROVED** [review]

**Pull request merged**